sqlitecluster/SQLiteNode.cpp

```diff
@@ -1656,33 +1653,43 @@ void SQLiteNode::_onMESSAGE(SQLitePeer* peer, const SData& message) {
             if (_replicationThreadsShouldExit) {
                 SINFO("Discarding replication message, stopping FOLLOWING");
             } else {
-                auto threadID = _replicationThreadCount.fetch_add(1);
-                SDEBUG("Spawning concurrent replicate thread (blocks until DB handle available): " << threadID);
-                try {
-                    uint64_t threadAttemptStartTimestamp = STimeNow();
-                    _replicateThreadStarted = false;
-                    thread(&SQLiteNode::_replicate, this, peer, message, _dbPool->getIndex(false), threadAttemptStartTimestamp).detach();
-                    {
-                        unique_lock<mutex> lock(_replicateStartMutex);
-                        while (!_replicateThreadStarted) {
-                            _replicateStartCV.wait(lock);
-                            if (!_replicateThreadStarted) {
-                                SINFO("condition variable finished waiting but replicate thread not started.");
+                if (SIEquals(message.methodLine, "COMMIT_TRANSACTION")) {
+                        // For COMMIT_TRANSACTION messages, we do not start a new thread. This aoids a race condition where we could spin up the
+                        // COMMIT thread, but not yet have called `_leaderCommitNotifier.notifyThrough` for the current transaction number while
+                        // the sync thread changes states. Particularly, if the sync thread drops out of FOLLOWING before this happens,
+                        // It may mean that we drop commits that leader had sent us, because we haven't recorded that we received them.
+                        // When leader is standing down this can ultimately lead to a fork.
```

Comment:
I think this is explaining the previous behavior, right? The way it's written, I felt like it could cause a fork after these changes